trailmap/acquisition_paths.py

```diff
@@ -0,0 +1,59 @@
+import networkx as nx
+from pprint import pprint
+import trailmap.pathway_analysis as pa
+
+
+def conduct_apa(facility_dict_in, facility_dict_out):
+    ''' '''
+    G = build_graph(facility_dict_in, facility_dict_out)
+
+    sources = [key for (key, value) in facility_dict_in.items() if not value]
+    sinks = list(node for node, out_deg in G.out_degree() if out_deg == 0)
+
+    print("\nSimple paths")
+    pathways = find_simple_paths(G, sources, sinks)
+    print(pathways)
+
+    return G, pathways
+
+
+def build_graph(facility_dict_in, facility_dict_out):
+    '''Builds NetworkX graph from Cyclus input file
+    '''
+    G = nx.DiGraph()
+
+    for facility in facility_dict_in.keys():
+        G.add_node(facility)
+    for facility in facility_dict_in:
+        for incommod in facility_dict_in[facility]:
```

Comment:
I like that suggestion, thanks

Reply:
The loops are different because they're doing different things. For the receiver, you're iterating through every incommod in the list of incommods to be able to find senders that can provide that commod. But when you're looping through senders, you just need to check that the commod is in the list oucommods. You don't need to iterate through outcommods, because you don't care about the other outcommods So even if I changed the loop, I would still need to iterate over incommods directly later to get individual commods. That's why I wrote it the way I did

I'm sure there's a better way of doing this, but I'm not sure your suggestion is better per sae because it won't reduce the number of nested loops